internal/cri/store/container/status.go

```diff
@@ -164,21 +166,54 @@ type StatusStorage interface {
 // StoreStatus creates the storage containing the passed in container status with the
 // specified id.
 // The status MUST be created in one transaction.
+// If atomic writing fail due to no space, try use the in-place modification method, see issue 7247 for details.
 func StoreStatus(root, id string, status Status) (StatusStorage, error) {
 	data, err := status.encode()
 	if err != nil {
 		return nil, fmt.Errorf("failed to encode status: %w", err)
 	}
 	path := filepath.Join(root, "status")
 	if err := continuity.AtomicWriteFile(path, data, 0600); err != nil {
-		return nil, fmt.Errorf("failed to checkpoint status to %q: %w", path, err)
+		if strings.Contains(err.Error(), "no space") {
```

Comment:
I filed a fix plan in https://github.com/containerd/containerd/issues/11504.

Please take a look on that.

My idea is to introduce a new `StoreStatus` which is used in restart.go file only.
In your patch, in place doesn't make sense to me. We can't guarantee it always work.
Based on that, we can just ignore no-space error and use memory status directly.

Reply:
looks better and has been modified.